tokio-quiche/tests/integration_tests/headers.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2025, Cloudflare, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::fixtures::*;
+
+use futures::SinkExt;
+
+use tokio_quiche::http3::driver::H3Event;
+use tokio_quiche::http3::driver::IncomingH3Headers;
+use tokio_quiche::http3::driver::OutboundFrame;
+use tokio_quiche::http3::driver::ServerH3Event;
+use tokio_quiche::quiche::h3::Header;
+
+#[tokio::test]
+async fn test_additional_headers() {
+    let hook = TestConnectionHook::new();
+
+    let url = start_server_with_settings(
+        QuicSettings::default(),
+        Http3Settings::default(),
+        hook,
+        move |mut h3_conn| async move {
+            let event_rx = h3_conn.h3_controller.event_receiver_mut();
+
+            while let Some(frame) = event_rx.recv().await {
+                let ServerH3Event::Core(frame) = frame;
+
+                match frame {
+                    H3Event::IncomingHeaders(headers) => {
+                        let IncomingH3Headers { mut send, .. } = headers;
+
+                        // Send initial headers.
+                        send.send(OutboundFrame::Headers(vec![Header::new(
+                            b":status", b"103",
+                        )]))
+                        .await
+                        .unwrap();
+
+                        tokio::task::yield_now().await;
```

Comment:
small nit: may be worth leaving a comment as to why the yield is necessary

Reply:
It's not really necessary TBH and I just copied it from the other test we have, but it seems maybe better to avoid sending the frames back to back, just to make sure it still works with a bit of delay.